### PR TITLE
feat: create actions

### PR DESCRIPTION
**Pull request by prousso**

### CODE REVIEW

File: requirements.in [hunk 1]
```diff
@@ -0,0 +1 @@
+PyYAML
```
Comment (tleveque69) [inline requirements.in:1]: Why do we need both dev and requirements it seems like the recipe requirements isn't really called

File: src/ga_actions_check_ref.py [hunk 2]
```diff
@@ -42,7 +47,7 @@ def is_github_workflow_valid(file: IO, allowed_actions: Mapping[str, str]) -> bo
 
 
 def load_allowed_actions() -> Mapping[str, str]:
-    with open("ALLOWED_ACTIONS.yaml", mode="r") as file:
+    with open("./ALLOWED_ACTIONS.yaml", mode="r") as file:
```
Comment (tleveque69) [inline src/ga_actions_check_ref.py:50]: We use a relative path here but the file isn't alway there

File: src/ga_actions_check_ref.py [hunk 3]
```diff
@@ -2,37 +2,42 @@
...
 
+import yaml
+
+ALLOWED_ORGANIZATIONS = ["actions", "lumapps"]
 SHORT_SHA1_PATTERN = re.compile(
     r"^(?P<actions_name>[A-Za-z0-9_.-]*\/[A-Za-z0-9_.-]*)@(?P<actions_version>[a-f0-9]{7})$"
 )
 
 
+# pylint: disable=too-many-nested-blocks,too-many-branches
```
Comment (tleveque69) [inline src/ga_actions_check_ref.py:15]: This is hard to maintain / review can we not split it into sub function so we avoir ignoring pylint

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+sync: venv
+	pip-sync dev-requirements.txt
+
+venv:
+	python3.7 -m venv venv
+	pip install --quiet --upgrade pip
+	pip install --quiet pip-tools
+
+requirements.txt: venv requirements.in
```
Comment (tleveque69) [inline Makefile:17]: This recipe is only call by dev requirement do we really need to separate them if not used by it's own

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v2.4.0
+  hooks:
+  - id: trailing-whitespace
+    exclude: .github/CODEOWNERS
+    stages: ["commit"]
+  - id: end-of-file-fixer
+    exclude_types: [json]
```
Comment (tleveque69) [inline .pre-commit-config.yaml:17]: Why do we exclude json files ?

File: src/ga_actions_check_ref.py (hunk 2 above)
Comment (prousso) [reply src/ga_actions_check_ref.py:72]: We must use relative path to call it in action

File: Makefile (hunk 4 above)
Comment (prousso) [reply Makefile:17]: Never call but useful when use some dependencies

**Review by tleveque69 — APPROVED** [review]
Would prefer to have create a TS github action to use the github action library but this LGTM

**Pull request merged**